app/models/spree/product_decorator.rb

```diff
@@ -275,19 +275,23 @@ def ensure_standard_variant
   # This fixes any problems arising from failing master saves, without the need for a validates_associated on
   # master, while giving us more specific errors as to why saving failed
   def save_master
-    begin
-      if master && (master.changed? || master.new_record? || (master.default_price && (master.default_price.changed? || master.default_price.new_record?)))
-        master.save!
-      end
+    if master && (
+        master.changed? || master.new_record? || (
+          master.default_price && (
+            master.default_price.changed? || master.default_price.new_record?
+          )
+        )
+      )
+      master.save!
+    end
 
     # If the master cannot be saved, the Product object will get its errors
     # and will be destroyed
-    rescue ActiveRecord::RecordInvalid
-      master.errors.each do |att, error|
-        self.errors.add(att, error)
-      end
-      raise
+  rescue ActiveRecord::RecordInvalid
+    master.errors.each do |att, error|
+      self.errors.add(att, error)
     end
+    raise
```

Comment:
welcome to spree v3 where this is done: https://github.com/spree/spree/commit/fe6369f9d289d7d69b0970b576d77f41a1eb7061